consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/validator/ForkingValidatorProvider.java

```diff
@@ -71,46 +70,32 @@ public ForkingValidatorProvider(
 
   private Collection<Address> getValidators(
       final long block, final Function<ValidatorProvider, Collection<Address>> getValidators) {
-    final Optional<ValidatorSelectorConfig> fork = forksSchedule.getFork(block);
+    final BftForkSpec<QbftConfigOptions> forkSpec = forksSchedule.getFork(block);
     final ValidatorProvider validatorProvider = resolveValidatorProvider(block);
 
-    if (fork.isPresent()) {
-      final VALIDATOR_SELECTION_MODE validatorSelectionMode =
-          fork.get().getValidatorSelectionMode();
-
-      // when moving to a block validator the first block needs to be initialised or created with
-      // the previous block state otherwise we would have no validators
-      if (validatorSelectionMode.equals(BLOCKHEADER)) {
-        if (block > 0 && block == fork.get().getBlock()) {
-          final long prevBlockNumber = block - 1L;
-          final Optional<BlockHeader> prevBlockHeader = blockchain.getBlockHeader(prevBlockNumber);
-          if (prevBlockHeader.isPresent()) {
-            return resolveValidatorProvider(prevBlockNumber)
-                .getValidatorsForBlock(prevBlockHeader.get());
-          }
+    // when moving to a block validator the first block needs to be initialised or created with
+    // the previous block state otherwise we would have no validators
+    if (forkSpec.getConfigOptions().getValidatorContractAddress().isEmpty()) {
```

Comment:
This check feels slightly less obvious than before.
Not a blocker, just a suggestion:
Add isBlockHeaderMode() (and isValidatorContractMode()?) for clarity?
Or maybe validatorSelectionMode.equals(BLOCKHEADER) as before?